let main =
  if (failure 0) then {
    print error_message
  } else {
    exit EXIT_FAILURE
  }